src/rust/socket/listener.rs

```diff
@@ -702,9 +702,23 @@ impl Listener {
                 request = request.header("Cookie", cookie);
             }
         }
+        let connector = match url.scheme() {
+            "ws" => {
+                Some(tokio_tungstenite::Connector::Plain)
+            },
+            "wss" => {
+                let tls_con = native_tls::TlsConnector::new().map_err(|e| (ConnectError::from(e), reconnect))?;
```

Comment:
This should ideally support inserting attributes about a self-signed cert. For now, I verified this works on watchOS,tvOS,iOS, android and macOS by adding:
```rust
let mut builder = TlsConnector::builder();
builder.danger_accept_invalid_certs(true);
builder.danger_accept_invalid_hostnames(true);
builder.disable_built_in_roots(true);
let tls_con = builder.build().map_err(|e| (ConnectError::from(e), reconnect))?
```